File: Practica2/Circulo.java
package Practica2;

public class Circulo {
    public static class Punto {
        public int x;
        public int y;

        public Punto(int xValor, int yValor) {
            x = xValor;
            y = yValor;
        }
        @Override
        public String toString() {
            return x + ", " + y;
        }
    }

    public Punto centro;
    public float radio;

    public Circulo(Punto c, float r) {
        centro = c;
        radio = r;
    }
    @Override
    public String toString() {
        return "Circulo(" + centro + ", radio=" + radio + ")";
    }

    public void dibujaCirculo() {
        System.out.println("Dibujando un círculo en " + centro + " con radio " + radio);
    }

    public static void main(String[] args) {
        Circulo.Punto centro = new Circulo.Punto(5, 10);
        Circulo circulo = new Circulo(centro, 3.5f);
        System.out.println(circulo);
        circulo.dibujaCirculo();
    }
}

File: Practica2/Circulo.py
class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def __str__(self):
        return "{}, {}".format(self.x, self.y)

class Circulo:
    def __init__(self, centro, radio):
        self.centro = centro
        self.radio = radio

    def __str__(self):
        return "Circulo({}, {}, radio={})".format(self.centro.x, self.centro.y, self.radio)

    def dibuja_circulo(self):
        print("Dibujando un círculo en {} con radio {}".format(self.centro, self.radio))

centro = Punto(5, 10)
circulo = Circulo(centro, 3.5)
print(circulo)
circulo.dibuja_circulo()